### ISSUE TITLE
Hot Reload with dotnet watch run doen't work. The call /_framework/blazor-hotreload returns status 204

### ISSUE THREAD

**Issue by xrkolovos** [dotnet/aspnetcore#36745]
### Describe the bug
I have updated my project to .net 6 rc1 and I am trying to test hotreload. I use dotnet watch run. In the basic template sample application it works in my pc. In my project it doesn't work. 

I get the message
`watch : Hot reload enabled. For a list of supported edits, see https://aka.ms/dotnet/hot-reload. Press "Ctrl + Shift + R" to restart.`
but when the browser starts the http call /_framework/blazor-hotreload returns status 204. In the sample app it returns 200 and it works.

### Exceptions (if any)
I can't find any errors - Exceptions.

### Further technical details
ASP.NET Core version 6.0.100-rc.1

.NET SDK (reflecting any global.json):
 Version:   6.0.100-rc.1.21458.32
 Commit:    d7c22323c4

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-rc.1.21458.32\

Host (useful for support):
  Version: 6.0.0-rc.1.21451.13
  Commit:  d7619cd4b1

.NET SDKs installed:
  5.0.100 [C:\Program Files\dotnet\sdk]
  5.0.401 [C:\Program Files\dotnet\sdk]
  6.0.100-rc.1.21458.32 [C:\Program Files\dotnet\sdk]

**Comment by xrkolovos** [comment]
Is there any suggestions where to look for error logs?
Thank you

**Comment by xrkolovos** [comment]
Hello

Do you need any help with this one? Any suggestions how to investigate it?

Thank you

**Comment by pranavkm** [comment]
Is there a minimal app that reproduces this issue consistently? That might be helpful. Also, are you able to try this using VS 17.0 Preview 4.1? It supports hot reload for blazor-webassembly apps when you run without the debugger attached (Ctrl+F5)?

**Comment by xrkolovos** [comment]
> Is there a minimal app that reproduces this issue consistently? 

The basic blazor startup template from visual studio now has the same issue. I get 204 in this call.

It never get's inside here because of 204.

![image](https://user-images.githubusercontent.com/517802/134799719-e42b0dbf-8bff-4a69-a189-3e7d0a72231d.png)

> Also, are you able to try this using VS 17.0 Preview 4.1? It supports hot reload for blazor-webassembly apps when you run without the debugger attached (Ctrl+F5)?

Yes I tried it and I have the same issue


Is there any logs from the Hot Reload system?

**Comment by xrkolovos** [comment]
I tried reinstalling net 6 rc1 sdk and also deleting .nuget cache folder. I still have this issue.

**Comment by xrkolovos** [comment]
more info with all LogLevels in Trace

![image](https://user-images.githubusercontent.com/517802/134800929-1fb6fc24-2efb-4cd5-9103-653a2d39f981.png)

logs
```
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 GET https://localhost:7015/_framework/blazor-hotreload - -
trce: Microsoft.AspNetCore.Server.Kestrel.Http2[49]
      Connection id "0HMC0PC7THGMN" sending PING frame for stream ID 0 with length 8 and flags ACK.
trce: Microsoft.AspNetCore.Server.Kestrel.Http2[49]
      Connection id "0HMC0PC7THGMN" sending HEADERS frame for stream ID 5 with length 33 and flags END_STREAM, END_HEADERS.
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished HTTP/2 GET https://localhost:7015/_framework/blazor-hotreload - - - 204 - - 3.7397ms
```

**Comment by mkArtakMSFT** [comment]
This has been fixed as part of 6.0-rc2 release.
